### PR TITLE
Refactor the logging

### PR DESCRIPTION
**Pull request by mraspaud**
This PR refactors the logging in trollmoves and provides a couple of new functions.
As an asumption, I consider the log config as the way to go in the future and hence issue a deprecation warning when using the old method with -l and -v (although they still work).

 - [x] Tests added  
 - [x] Fully documented

### CODE REVIEW

**Review by pnuu — COMMENTED** [review]
One comment inline. And should we have example logging config(s) in `examples/` directory?

Otherwise LGTM

**Comment by mraspaud** [comment]
> And should we have example logging config(s) in `examples/` directory?

Well the logging config format is the one given by the logging library. How about I just add a link to the format documentation in the trollmoves documentation?

File: trollmoves/logging.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import pathlib
+import warnings
+from contextlib import suppress
+
+import yaml
+
+
+def add_logging_options_to_parser(parser, legacy=False):
+    """Add logging options to parser."""
+    if not legacy:
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply trollmoves/logging.py:33]: oh, I thought I did that, sorry. Looking into it now.

[the remark this answers is not in the document]
Comment (mraspaud) [reply trollmoves/logging.py]: @pnuu Added a test for it and fixed.

**Pull request merged**